src/gen_rules.ml

```diff
@@ -138,6 +138,16 @@ module Gen(P : Params) = struct
       | None -> build
       | Some f -> Build.fail f >>> build
 
+    let interpret_lib_deps_for_dot_merlin ~dir lib_deps =
+      let internals, externals, _ = Lib_db.interpret_lib_deps t ~dir lib_deps in
```

Comment:
The third argument is used in case of error, for instance if a dependency is missing. The `fail` value can be passed to `Build.fail`, in order to fail only when we execute the rule instead of when we generate the rules. Although this function shouldn't be needed, see comments in `requires_and_dot_merlin`